### ISSUE TITLE
[Clover] Document (or add) how to set HTTP Headers on the response

### ISSUE THREAD

**Issue by damienwebdev** [angular/universal#2749]
# 🚀 Feature request

#### What modules are relevant for this feature request?

- [ ] builders
- [x] common
- [ ] express-engine

### Description

I'm trying to bring Varnish into my SSR Pipeline in order to improve TTFB. In order to do this, I need to be able to set HTTP Headers on the response from inside of my Angular app at SSR Time. 

### Describe the solution you'd like

I'd like to make my Angular SSR app Cacheable (And cache-bustable) by leveraging HTTP Headers when using Clover.

Currently the API looks like

```ts
export interface Engine = {
    render(options: RenderOptions): Promise<string>;
}
```

Given that Clover is still experimental, maybe we could slightly adjust the API with

```ts
export interface RenderOutput {
  // The rendered HTML Output
  html: string

  // Data made specifically available for output after rendering.
  data: Record<string | number | symbol, unknown>;
}

export interface Engine {
  render(options: RenderOptions): Promise<RenderOutput>;
}
```

### Describe alternatives you've considered

I was taking a look at the `Engine` API and I didn't see anything that would help.

**Comment by geometricpanda** [comment]
I dont believe express allows us to change the HTTP status code after the render
However is it worth considering using a Resolver/Guard which uses:
`@Inject(RESPONSE) private response: Response`

then later
```
this.response.status(200);
```

**Comment by damienwebdev** [comment]
@geometricpanda this only works in the express engine. It doesn't work with Clover :( as the tokens are not supported (as far as I can tell).

**Comment by geometricpanda** [comment]
Does clover have any similar tokens?
